connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoaderTest.java

```diff
@@ -129,4 +133,57 @@ public void testLoadingMixOfValidAndInvalidPlugins() throws Exception {
             assertNotNull(classLoader.pluginClassLoader(pluginClassName));
         }
     }
+
+    @Test
+    public void testPluginConflictSameClassSameVersion() {
+        DelegatingClassLoader delegatingClassLoader = new DelegatingClassLoader(Collections.emptyList());
+
+        Class<Connector> klass = Connector.class;
+        PluginDesc<Connector> connectorDesc1 = new PluginDesc<>(
+                klass,
+                "1.1.0",
+                ClassLoader.getSystemClassLoader()
+        );
+        delegatingClassLoader.addPlugins(Arrays.asList(connectorDesc1, connectorDesc1), ClassLoader.getSystemClassLoader());
+        assertTrue(delegatingClassLoader.reportPluginConflicts().contains(klass.getName()));
+    }
+
+    @Test
+    public void testPluginConflictSameClassDiffVersions() {
+        DelegatingClassLoader delegatingClassLoader = new DelegatingClassLoader(Collections.emptyList());
+        PluginDesc<Converter> notConflictingPlugin = new PluginDesc<>(
+                Converter.class,
+                "0.0",
+                ClassLoader.getSystemClassLoader()
+        );
+        delegatingClassLoader.addPlugins(Arrays.asList(notConflictingPlugin), ClassLoader.getSystemClassLoader());
+
+        Class<Connector> klass = Connector.class;
+        PluginDesc<Connector> connectorDesc1 = new PluginDesc<>(
+                klass,
+                "1.1.0",
+                ClassLoader.getSystemClassLoader()
+        );
+        delegatingClassLoader.addPlugins(Arrays.asList(connectorDesc1), ClassLoader.getSystemClassLoader());
+        assertTrue(delegatingClassLoader.reportPluginConflicts().isEmpty());
+
+        PluginDesc<Connector> connectorDesc2 = new PluginDesc<>(
+                klass,
+                "1.0.0",
+                ClassLoader.getSystemClassLoader()
+        );
+        delegatingClassLoader.addPlugins(Arrays.asList(connectorDesc2), ClassLoader.getSystemClassLoader());
+        String pluginClassName = klass.getName();
+        assertTrue(delegatingClassLoader.reportPluginConflicts().contains(pluginClassName));
```

Comment:
It's probably worthwhile to also assert that the non-conflicting plugin has not accidentally been added as a conflict, here and after the final `addPlugins(...)` call below.

Reply:
done